pytest_textual_snapshot.py

```diff
@@ -33,6 +34,18 @@ class SVGImageExtension(SingleFileSnapshotExtension):
     _file_extension = "svg"
     _write_mode = WriteMode.TEXT
 
+    def _read_snapshot_data_from_location(self, *args, **kwargs) -> Optional["SerializableData"]:
+        """Normalize SVG data right after they are loaded from persistent storage."""
+        data = super()._read_snapshot_data_from_location(*args, **kwargs)
+        if data is not None:
+            data = normalize_svg(data)
+        return data
```

Comment:
I'm not sure about this. Subclassing a private method from a non-pinned 3rd party library is likely to eventually break if the method is removed or renamed.

Reply:
I agree, but I simply obeyed this:
https://github.com/syrupy-project/syrupy/blob/ef8189c68460593fead9c484405b755e272c8cca/src/syrupy/extensions/base.py#L140C10-L140C43

Reply:
@darrenburns So, what do you think?
By the way, we could pin pytest_textual_snapshot as an extra security measure.

Reply:
Sorry!

I think it makes sense to pin syrupy and merge this. Although perhaps longer term it'd make sense to have an option in Textual to write SVGs without the IDs rather than stripping them out.

Reply:
> Sorry!

No problem, the week before the 1.0.0 release must have been quite busy.
 
> I think it makes sense to pin syrupy and merge this.

> Although perhaps longer term it'd make sense to have an option in Textual to write SVGs without the IDs rather than stripping them out.

Stricto sensu, that option already exists... but it cannot be leveraged from pytest-textual-snapshot.

As part of my personal reflections following this PR, I think the ideal situation is to introduce a specialized, diff-friendly format that represents a TUI screenshot and can be turned into an SVG picture. That new format would be versioned into Git instead of the SVG pictures whereas the SVG pictures would still appear in the HTML report. I started imagining a simple text format based on _layers_ (background colors, text, foreground colors) but quickly noticed it lacks an elegant way to embed text formatting like bold, underline, etc. -- should this become a fourth layer or should it be shoehorned into the other layers? In the end, this endeavour is a significant amount of work for a rather small improvement, at a time when Textual users expect improvements in other areas (e.g. more widgets). So... yeah, definitely a longer term idea.